src/SSHDebugPS/Utilities/VSMessageBoxHelper.cs

```diff
@@ -0,0 +1,36 @@
+﻿using Microsoft.VisualStudio.Shell;
+using Microsoft.VisualStudio.Shell.Interop;
+using ThreadingTasks = System.Threading.Tasks;
+
+namespace Microsoft.SSHDebugPS.Utilities
+{
+    internal class VSMessageBoxHelper
+    {
+        /// <summary>
+        /// Switches to main thread and shows a VS message box.
+        /// </summary>
+        public static void ShowMessage(string title, string message, OLEMSGICON icon, OLEMSGBUTTON button, OLEMSGDEFBUTTON defaultButton)
```

Comment:
nit: I might suggest renaming this to something that makes it clear that the message box will be put up async. Maybe `PostUIMessage`.